src/Loess.jl

```diff
@@ -135,12 +141,11 @@ end
 # Returns:
 #   A length n' vector of predicted response values.
 #
-function predict(model::LoessModel{T}, z::T) where T <: AbstractFloat
-    predict(model, T[z])
+function predict(model::LoessModel, z::Number)
+    predict(model, [z])
```

Comment:
Could we move the algorithm below for `zs::AbstractVector` to an internal function `_predict(::LoessModel, ::Union{AbstractVector,Tuple})` that can be called with a `Tuple` as well (or just change the signature of `predict` below but this seems a bit weird maybe)? From a quick look at the implementation it seems that maybe the only additional change needed would be to relax `traverse(::KDTree, ::AbstractVector)` in the same way.
```suggestion
    predict(model, (z,))
```

Reply:
An alternative, of course, would be to implement the special scalar case separately but then code would be a bit duplicated.

Reply:
I plan to make optimizations along those lines in a follow-up PR.